7. Dynamic Programming/cookie_collect/[CPP]cookie_collect_csubhradipta.cpp

```diff
@@ -17,26 +18,28 @@ using namespace std;
 void solve(){
     int n;
     cin>>n;
-    int cookies[n+2][n], maxCookies[n];
-    for (int i = 0; i < n+2; i++){
-        for (int j = 0; j < n; j++){
-            
-            if(i >= 1 && i <= n)
-                cin>>cookies[i][j];
-            else
-                cookies[i][j] = 0;              // Initializing the top and bottom of matrix with row of zeroes
-        }                                   
+    int cookies[n][n], maxCookies[n];
+    for (int i = 0; i < n; i++){
+        for (int j = 0; j < n; j++)
+            cin>>cookies[i][j];       
     }
     
     
     for (int j = 1; j < n; j++) { 
-        for (int i = 1; i <= n ; i++) {         // Traversing matrix top to bottom
+        for (int i = 0; i < n ; i++) {          // Traversing matrix top to bottom
+                                                                                    // adding the max element in the previous column
+        	if ( i == 0)
+        		cookies[i][j] += max(cookies[i][j - 1], cookies[i + 1][j - 1]);                 // for top row, left-up element is not there
 
-                cookies[i][j] += max({cookies[i - 1][j - 1], cookies[i][j - 1], cookies[i + 1][j - 1]});    // adding the max element in the previous column
-                                                                                            // (left-up, left, left-down)
+        	else if ( i == n-1)
+        		cookies[i][j] += max(cookies[i - 1][j - 1], cookies[i][j - 1]);                         // for bottom row, left-down element is not there
 
-                if( j == n-1)
-                    maxCookies[i-1] = cookies[i][j];                                            // storing the elements of last column 
+        	else
+                cookies[i][j] += max({cookies[i - 1][j - 1], cookies[i][j - 1], cookies[i + 1][j - 1]});                // for intermediate rows, all element present
+                                                                                                            
+
+            if( j == n-1)
+                maxCookies[i] = cookies[i][j];                                  // storing the elements of last column 

```

Comment:
You're still using this array.

Reply:
> You're still using this array.

I thought you're asking for the row of zeroes.
Code updated.